rust/core-lib/src/event_context.rs

```diff
@@ -494,6 +494,22 @@ impl<'a> EventContext<'a> {
         self.editor.borrow_mut().dec_revs_in_flight();
     }
 
+    /// Checks a newline needs to be added to the end of the file
+    pub(crate) fn pre_save_newline(&mut self) -> Rope {
```

Comment:
given that this is now returning the rope we will save, I'd call it something like `text_for_save`. The comment is also no longer up to date; it isn't really a check, it just returns the text to be saved, appending a newline if necessary.